JVM Init Error: realpath: command not found

```bash
jvm init
/usr/local/bin/jvm: line 10: realpath: command not found
mv: cannot stat '/root/.juliarc.jl': No such file or directory
ERROR: could not open file /test-project/init
 in include at ./boot.jl:261
 in include_from_node1 at ./loading.jl:304
 in process_options at ./client.jl:308
 in _start at ./client.jl:411
mv: cannot stat '/root/.juliarc.jl.bak': No such file or directory
make: *** [.jdeps] Error 1
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,0 +1,16 @@
+
+language: julia
+os:
+  - linux
+  - osx
+julia:
+  - 0.3
+  - 0.4
+  - nightly
+notifications:
+  email: false
+script:
+  - if [[ -a .git/shallow ]]; then git fetch --unshallow; fi
+  - julia --check-bounds=yes -e 'Pkg.clone(pwd());'
+  - export PATH=`pwd`/bin:$PATH
+  - jvm test JVM

--- a/README.md
+++ b/README.md
@@ -1,45 +1,58 @@
 JVM: Julia Version Manager
+[![Build Status](https://travis-ci.org/tmlbl/JVM.jl.svg?branch=travis)](https://travis-ci.org/tmlbl/JVM.jl)
 ==========================
 
-The JVM is a tool for creating virtual environments in Julia and publishing
-Docker images based on them. To use it, you will want to have certain programs
-already installed, including:
+The JVM is a tool for creating and working with virtual environments in Julia.
+JVM integrates with Docker to generate images for Julia applications.
+
+## Requirements
 
 * A reasonably recent version of Julia
 * Tools necessary for Julia package installation: C compilers and tools for building
 software, like those covered by the build-essential Debian repository
-* Docker, for building images and launching containers
-* docker-squash, a pip package for removing unnecessary files from Docker images
+* Docker, for building images and launching containers (optional)
+* docker-squash, a pip package for removing unnecessary files from Docker images (optional)
 
-## Configuring
+## Creating a project
 
-A new project can be created in the current directory by running `jvm init`.
-This creates a basic configuration file, with the following properties:
+Run `jvm init` to create a project in the current directory. This creates a
+JSON manifest file. You can then add dependencies like so:
 
-```javascript
-{
-  "deps": {
-    "AppConf": "0.1.1", // Map of dependencies to versions
-    // Can do unregistered packages as well
-    "https://github.com/tmlbl/Oanda.jl.git": "d32e3d9ee2a867cb5f7093bf8d7d8eecf5160b0d"
-  },
-  // Version of Julia the project will use to run (installed automatically)
-  "julia": "0.4.5",
-  // Project name and name of resulting Docker image
-  "name": "jvm",
-  "scripts": {
-    // This section can contain arbitrary keys with bash commands, run with jvm [cmd]
-    "bootstrap": "jvm run scripts/bootstrap.jl",
-    // If the value is the file, the file will be executed
-    "test": "test/runtests.jl"
-  },
-  // This is the version of the project itself
-  // Docker images will be tagged with this version number
-  "version": "0.0.1",
-  // Here arbitrary lines can be injected before and after the package build
-  // step when the image is created.
-  "pre-build": [
-    "RUN apt-get install libsnappy1"
-  ]
-}
+```bash
+$ jvm add StatsBase
+$ jvm add HttpServer 0.1.4
 ```
+
+When a project contains this manifest file, you can fetch the required packages
+by running `jvm install`.
+
+## Running commands and files
+
+Running `jvm` alone will open a REPL. Here is where you should manage your
+package versions. To save a state in the manifest, run `jvm freeze`.
+
+Use `jvm run` to evaluate Julia files in the local context.
+
+## Creating a Docker image
+
+Create a Dockerfile in your project that uses `${your_project}-base` as the
+base image. JVM builds packages and generates cache files in a separate step
+to speed up build times. Running `jvm image` will kick off the build.
+
+Additional lines can be added to the base image in the manifest file like so:
+
+```json
+
+"pre-build": [
+  "RUN apt-get install libsnappy1"
+],
+"post_build": [
+  "WORKDIR /opt"
+]
+```
+
+## Running tests
+
+Running `jvm test ${pkg_name}` will run `Pkg.test` on the default version of
+Julia in a sandbox environment for the given package. Another Julia version can
+be specified like so: `jvm test JVM 0.4.0`. This will probably change.

--- a/REQUIRE
+++ b/REQUIRE
@@ -1,3 +1,2 @@
-julia 0.4-
 JSON
 Mustache

--- a/bin/jvm
+++ b/bin/jvm
@@ -1,23 +1,29 @@
 #!/bin/bash
 
+ARGS="$@"
 OS=`uname`
 
-ARGS="$@"
-
+# Get cli.jl location
 if [ "$OS" = "Darwin" ]; then
   SCRIPT=`readlink ${0} | awk '{gsub(/jvm/,"cli.jl");}1'`
 else
   SCRIPT=`readlink -f ${0} | awk '{gsub(/jvm/,"cli.jl");}1'`
 fi
 
+# Move juliarc.jl out of the way temporarily
+# TODO: Better way to handle this
 JULIA_RC=$HOME/.juliarc.jl
 
 if [[ -e "$JULIA_RC" ]]; then
-  mv $JULIA_RC $JULIA_RC.bak
+  mv $JULIA_RC $JULIA_RC.jvm.bak
 fi
 
+# Launch JVM package in julia
 julia -q --color=yes $SCRIPT $ARGS
+xit=$?
 
-if [[ -e "$JULIA_RC.bak" ]]; then
-  mv $JULIA_RC.bak $JULIA_RC
+if [[ -e "$JULIA_RC.jvm.bak" ]]; then
+  mv $JULIA_RC.jvm.bak $JULIA_RC
 fi
+
+exit $xit

--- a/bin/jvm
+++ b/bin/jvm
@@ -1,23 +1,35 @@
 #!/bin/bash
 
 OS=`uname`
-
 ARGS="$@"
 
-if [ "$OS" = "Darwin" ]; then
-  SCRIPT=`readlink ${0} | awk '{gsub(/jvm/,"cli.jl");}1'`
+if [ ! -h $0 ]; then
+  a1="${0}"
 else
-  SCRIPT=`readlink -f ${0} | awk '{gsub(/jvm/,"cli.jl");}1'`
+  # Get cli.jl location
+  if [ "$OS" = "Darwin" ]; then
+    a1=`readlink ${0}`
+  else
+    a1=`readlink -f ${0}`
+  fi
 fi
 
-JULIA_RC=$HOME/.juliarc.jl
+SCRIPT=${a1/jvm/cli.jl}
+
+# Move juliarc.jl out of the way temporarily
+# TODO: Better way to handle this
+JULIA_RC="$HOME/.juliarc.jl"
 
 if [[ -e "$JULIA_RC" ]]; then
-  mv $JULIA_RC $JULIA_RC.bak
+  mv $JULIA_RC $JULIA_RC.jvm.bak
 fi
 
+# Launch JVM with julia
 julia -q --color=yes $SCRIPT $ARGS
+xit=$?
 
-if [[ -e "$JULIA_RC.bak" ]]; then
-  mv $JULIA_RC.bak $JULIA_RC
+if [[ -e "$JULIA_RC.jvm.bak" ]]; then
+  mv $JULIA_RC.jvm.bak $JULIA_RC
 fi
+
+exit $xit

--- a/jvm.json
+++ b/jvm.json
@@ -9,7 +9,7 @@
     "AppConf": "0.1.1",
     "URIParser": "0.1.3"
   },
-  "julia": "0.4.5",
+  "julia": "0.3.10",
   "name": "jvm",
   "scripts": {},
   "pre-build": [

--- a/src/Config.jl
+++ b/src/Config.jl
@@ -5,7 +5,7 @@
 
 typealias SHA ASCIIString
 
-typealias DepVersion Union{SHA,VersionNumber}
+typealias DepVersion @compat Union{SHA,VersionNumber}
 
 type Dep
   name::AbstractString

--- a/src/JVM.jl
+++ b/src/JVM.jl
@@ -1,8 +1,10 @@
-__precompile__()
+isdefined(:__precompile__) && __precompile__()
 
 module JVM
 
-using JSON, Mustache
+using JSON,
+      Mustache,
+      Compat
 
 local_dir = ""
 package_dir = "/tmp/.jvm"
@@ -23,18 +25,14 @@ function localize()
   if isfile(CONFIG_FILE)
     global config = getconfig()
   end
+  ENV["JULIA_PKGDIR"] = local_dir
+  global JULIA_VERSION = "v$(VERSION.major).$(VERSION.minor)"
   run(`mkdir -p $(ENV["JULIA_PKGDIR"])`)
 end
 
 function __init__()
   global JULIA_PKGDIR_ORIG = Pkg.dir()
   global local_dir = joinpath(pwd(), ".jvm")
-  ENV["JULIA_PKGDIR"] = local_dir
-  global JULIA_VERSION = "v$(VERSION.major).$(VERSION.minor)"
-  # Hack to fix the library load path
-  Base.LOAD_CACHE_PATH[1] =
-    joinpath(local_dir, "lib/$JULIA_VERSION")
-
   update_env()
 end
 

--- a/src/cli.jl
+++ b/src/cli.jl
@@ -58,7 +58,7 @@ Commands:
     jvm              Start a REPL in the current environment
     jvm init         Create a new project in the current directory
     jvm install      Install dependencies for the current project
-    jvm test <pkg>   Run tests for <pkg> in current environment
+    jvm test         Run tests: Experimental
     jvm add <pkg>    Install a package or git repo and save it to jvm.json
     jvm freeze       Update jvm.json to match current state of project
     jvm package      Generate a tarball of assets ready for offsite installation

--- a/src/tasks.jl
+++ b/src/tasks.jl
@@ -1,3 +1,5 @@
+# EXPERIMENTAL
+# If a package contains a jvm.json, run its tests on multiple versions of julia
 function test(args)
   test_dir = "/tmp/.jdeps.test"
   if length(args) < 1

--- a/test/REQUIRE
+++ b/test/REQUIRE
@@ -1,0 +1,1 @@
+FactCheck

--- a/test/runtests.jl
+++ b/test/runtests.jl
@@ -1,9 +1,11 @@
 using JVM,
-      Base.Test
+      FactCheck
 
-@test !JVM.isgit("Compat")
-@test !JVM.isgit("AppConf")
-@test JVM.isgit("git@github.com:tmlbl/AppConf.jl")
-@test JVM.isgit("https://github.com/tmlbl/AppConf.jl")
-@test JVM.namefromgit("git@github.com:tmlbl/AppConf.jl.git") == "AppConf"
-@test JVM.namefromgit("https://github.com/tmlbl/AppConf.jl") == "AppConf"
+facts("git utils") do
+  @fact !JVM.isgit("Compat") --> true
+  @fact !JVM.isgit("AppConf") --> true
+  @fact JVM.isgit("git@github.com:tmlbl/AppConf.jl") --> true
+  @fact JVM.isgit("https://github.com/tmlbl/AppConf.jl") --> true
+  @fact JVM.namefromgit("git@github.com:tmlbl/AppConf.jl.git") --> "AppConf"
+  @fact JVM.namefromgit("https://github.com/tmlbl/AppConf.jl") --> "AppConf" 
+end

